framework/include/variables/MooseVariableFE.h

```diff
@@ -473,6 +477,13 @@ class MooseVariableFE : public MooseVariableField<OutputType>
   virtual void computeElemValues() override;
   virtual void computeElemValuesFace() override;
   virtual void computeFaceValues(const FaceInfo & fi) override;
+
+  /// Adding a adGradSln that is the average of the face gradients on both sides
+
+  VectorValue<ADReal> _ad_grad_face_avg;
+  void computeAdGradFaceAvg(const FaceInfo & fi);
+  VectorValue<ADReal> adGradSln(const FaceInfo & fi);
```

Comment:
We also generally try to put methods before data. Additionally data should tend to be less accessible than methods. For instance I suspect that `_ad_grad_face_avg` could probably be made private